vnext/Microsoft.ReactNative/Views/Image/ReactImage.cpp

```diff
@@ -312,30 +323,29 @@ winrt::fire_and_forget ReactImage::SetBackground(bool fireLoadEndEvent) {
       }
 
       if (source.sourceFormat == ImageSourceFormat::Svg) {
-        winrt::SvgImageSource svgImageSource{imageBrush.ImageSource().try_as<winrt::SvgImageSource>()};
+        winrt::SvgImageSource svgImageSource{};
 
-        if (!svgImageSource) {
-          svgImageSource = winrt::SvgImageSource{};
+        strong_this->m_svgImageSourceOpenedRevoker =
+            svgImageSource.Opened(winrt::auto_revoke, [weak_this, fireLoadEndEvent](const auto &, const auto &) {
+              auto strong_this{weak_this.get()};
+              if (strong_this && fireLoadEndEvent) {
```

Comment:
if `fireLoadEvent` is false, you can skip setting up this event delegate and that lets you not test this value here.